node/src/actors/chain_manager/mining.rs

```diff
@@ -939,14 +940,23 @@ fn build_block(
             tally_txns.push(ta_tx.clone());
             let commits_count = dr_state.info.commits.len();
             let reveals_count = dr_state.info.reveals.len();
-            let honests_count = commits_count - ta_tx.slashed_witnesses.len();
+
+            let (liars_count, errors_count) = calculate_liars_and_errors_count_from_tally(&ta_tx);
```

Comment:
This operation is only needed to calculate the `extra_tally_fee`. If we could refactor this function from accepting `tally_transactions: &[TallyTransaction]` into `tally_transactions_and_fees: &[(TallyTransaction, u64)]` we could skip this calculation.

But we would need to add that information to `create_tally_transactions`, and I'm not sure if this change will be small enough to be included in this PR :P

Reply:
uff, I want to merge as soon as possible, maybe we could open an issue to optimize some functions later

Reply:
That's fine